### PR TITLE
Refactor rosetta-client interface

### PR DESCRIPTION
**Pull request by Lohann**
## Description

Removes `method` and `function_selector` from rosetta-client interfaces, also accept raw bytes instead of string as parameter.

## Type of change
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: rosetta-client/src/tx_builder.rs [hunk 1]
```diff
@@ -26,31 +27,28 @@ impl GenericTransactionBuilder {
...
         amount: u128,
     ) -> Result<GenericMetadataParams> {
         Ok(match self {
-            Self::Astar(tx) => {
-                AstarMetadataParams(tx.method_call(contract, method, params, amount)?).into()
-            },
-            Self::Ethereum(tx) => tx.method_call(contract, method, params, amount)?.into(),
-            Self::Polkadot(tx) => tx.method_call(contract, method, params, amount)?.into(),
+            Self::Astar(tx) => AstarMetadataParams(tx.method_call(contract, data, amount)?).into(),
+            // Self::Ethereum(tx) => tx.method_call(contract, data, amount)?.into(),
```
Comment (dvc94ch) [inline rosetta-client/src/tx_builder.rs:43]: why is ethereum commented out?

File: rosetta-client/src/tx_builder.rs [hunk 2]
```diff
@@ -26,31 +27,28 @@ impl GenericTransactionBuilder {
...
+            Self::Polkadot(tx) => tx.method_call(contract, data, amount)?.into(),
         })
     }
 
     pub fn deploy_contract(&self, contract_binary: Vec<u8>) -> Result<GenericMetadataParams> {
         Ok(match self {
             Self::Astar(tx) => AstarMetadataParams(tx.deploy_contract(contract_binary)?).into(),
-            Self::Ethereum(tx) => tx.deploy_contract(contract_binary)?.into(),
+            // Self::Ethereum(tx) => tx.deploy_contract(contract_binary)?.into(),
```
Comment (dvc94ch) [inline rosetta-client/src/tx_builder.rs:51]: and here

File: rosetta-client/src/tx_builder.rs [hunk 3]
```diff
@@ -68,11 +66,11 @@ impl GenericTransactionBuilder {
...
-            (
-                Self::Ethereum(tx),
-                GenericMetadataParams::Ethereum(params),
-                GenericMetadata::Ethereum(metadata),
-            ) => tx.create_and_sign(config, params, metadata, secret_key),
+            // (
+            //     Self::Ethereum(tx),
+            //     GenericMetadataParams::Ethereum(params),
+            //     GenericMetadata::Ethereum(metadata),
+            // ) => tx.create_and_sign(config, params, metadata, secret_key),
```
Comment (dvc94ch) [inline rosetta-client/src/tx_builder.rs:73]: and here

File: rosetta-client/src/wallet.rs [hunk 4]
```diff
@@ -140,24 +144,24 @@ impl Wallet {
...
+    //     block_identifier: Option<PartialBlockIdentifier>,
+    // ) -> Result<serde_json::Value> {
+    //     let req = CallRequest {
+    //         network_identifier: self.client.config().network(),
+    //         method,
+    //         parameters: params.clone(),
+    //         block_identifier,
+    //     };
+    //     self.client.call(&req).await
+    // }
```
Comment (dvc94ch) [inline rosetta-client/src/wallet.rs:164]: fix or remove

File: rosetta-client/src/wallet.rs [hunk 5]
```diff
@@ -228,85 +232,159 @@ impl Wallet {
...
+            GenericClient::Astar(client) => client.call(&EthQuery::CallContract(call)).await?,
+            GenericClient::Polkadot(_) => anyhow::bail!("polkadot doesn't support eth_view_call"),
+            GenericClient::Bitcoin(_) => anyhow::bail!("bitcoin doesn't support eth_view_call"),
+        };
+        let EthQueryResult::CallContract(exit_reason) = result else {
+            anyhow::bail!("[this is a bug] invalid result type");
+        };
+        Ok(exit_reason)
+        // let method = format!("{contract_address}-{method_signature}-call");
+        // self.call(method, &json!(params), block_identifier).await
```
Comment (dvc94ch) [inline rosetta-client/src/wallet.rs:303]: remove

File: rosetta-client/src/wallet.rs [hunk 6]
```diff
@@ -228,85 +232,159 @@ impl Wallet {
...
+            },
+            GenericClient::Polkadot(_) => anyhow::bail!("polkadot doesn't support eth_storage"),
+            GenericClient::Bitcoin(_) => anyhow::bail!("bitcoin doesn't support eth_storage"),
+        };
+        let EthQueryResult::GetStorageAt(value) = result else {
+            anyhow::bail!("[this is a bug] invalid result type");
+        };
+        Ok(value)
+        // let method = format!("{contract_address}-{storage_slot}-storage");
+        // self.call(method, &json!({}), block_identifier).await
```
Comment (dvc94ch) [inline rosetta-client/src/wallet.rs:333]: remove

File: rosetta-client/src/wallet.rs [hunk 7]
```diff
@@ -228,85 +232,159 @@ impl Wallet {
...
+            GenericClient::Astar(client) => client.call(&EthQuery::GetProof(get_proof)).await?,
+            GenericClient::Polkadot(_) => anyhow::bail!("polkadot doesn't support eth_storage"),
+            GenericClient::Bitcoin(_) => anyhow::bail!("bitcoin doesn't support eth_storage"),
+        };
+        let EthQueryResult::GetProof(proof) = result else {
+            anyhow::bail!("[this is a bug] invalid result type");
+        };
+        Ok(proof)
+        // let method = format!("{contract_address}-{storage_slot}-storage_proof");
+        // self.call(method, &json!({}), block_identifier).await
```
Comment (dvc94ch) [inline rosetta-client/src/wallet.rs:362]: remove

File: rosetta-client/src/wallet.rs [hunk 8]
```diff
@@ -228,85 +232,159 @@ impl Wallet {
...
+            },
+            GenericClient::Polkadot(_) => anyhow::bail!("polkadot doesn't support eth_storage"),
+            GenericClient::Bitcoin(_) => anyhow::bail!("bitcoin doesn't support eth_storage"),
+        };
+        let EthQueryResult::GetTransactionReceipt(maybe_receipt) = result else {
+            anyhow::bail!("[this is a bug] invalid result type");
+        };
+        Ok(maybe_receipt)
+        // let call_method = format!("{}--transaction_receipt", hex::encode(tx_hash));
+        // self.call(call_method, &json!({}), None).await
```
Comment (dvc94ch) [inline rosetta-client/src/wallet.rs:388]: remove

**Review by dvc94ch — APPROVED** [review]
seems like this is way more changes than what we needed. fine with me, but it's a bit late for an indepth review.

File: rosetta-client/src/wallet.rs (hunk 4 above)
Comment (Lohann) [reply rosetta-client/src/wallet.rs:164]: We no longer need this one, removing..

File: rosetta-client/src/tx_builder.rs (hunk 1 above)
Comment (Lohann) [reply rosetta-client/src/tx_builder.rs]: Fixed, I was checking if I could add support to Polygon/Arbitrum without adding new items to the enum.

**Pull request merged**